### DISCUSSION TITLE
Input parameter validation

### DISCUSSION THREAD

**Discussion by michael-kopt** [CoreWCF/CoreWCF#1185] [Q&A] [answered]
Hi,

I'm moving a legacy project from WCF to CoreWCF.
In the old WCF project we have input parameter validation using annotations, like this:
<img width="871" alt="image" src="https://github.com/CoreWCF/CoreWCF/assets/81884813/58447f36-8240-47e5-8d7b-3f1c6e5c4cbf">

It was implemented using **SoapExtension** and **SoapExtensionAttribute** classes from **System.Web.Services.Protocols**, which no longer exist in .NET Core. Is it possible to implement something like this for CoreWCF service?

Thanks in advance.

**Comment by g7ed6e** [comment, marked as answer]
Hi @MikhailYugov 

An option would be to create your own ValidationAttribute and AssertAttribute matching the ones you are using from SoapExtension. Then implement and register with MSDI container an [IServiceBehavior responsible for adding a custom IDispatchMessageInspector to the DispatchRuntime MessageInspectors collection](https://learn.microsoft.com/en-us/dotnet/api/system.servicemodel.dispatcher.idispatchmessageinspector?view=dotnet-plat-ext-7.0).
The IDispatchMessageInspector AfterReceiveRequest method would reflect the current operation's attributes and then clone the Message to perform the xpath assertions.
